src/main/java/uk/gov/cshr/repository/InviteRepository.java

```diff
@@ -1,20 +1,28 @@
 package uk.gov.cshr.repository;
 
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.CrudRepository;
 import org.springframework.stereotype.Repository;
 import uk.gov.cshr.domain.Invite;
 import uk.gov.cshr.domain.InviteStatus;
 
+import javax.transaction.Transactional;
+import java.util.Optional;
+
 @Repository
 public interface InviteRepository extends CrudRepository<Invite, Long> {
 
     Invite findByForEmail(String forEmail);
 
     Invite findByCode(String code);
 
+    Optional<Invite> findByForEmailAndStatus(String code, InviteStatus status);
```

Comment:
Please check if Email and Status column of Invite db table has index, if indexes are not there then please add the indexes.

Reply:
sql file with creating index is added.